src/adapters/db.ts

```diff
@@ -2,17 +2,43 @@ import SQL from 'sql-template-strings'
 import { AppComponents } from '../types'
 
 export type IDatabaseComponent = {
-  getCommunityMembers: (communityId: string) => Promise<string[]>
+  getCommunityMembers: (
+    communityId: string,
+    options?: {
+      include?: string[]
+      exclude?: string[]
+    }
+  ) => Promise<string[]>
 }
 
-export async function createDBComponent(components: Pick<AppComponents, 'pg' | 'logs'>): Promise<IDatabaseComponent> {
-  const { pg, logs } = components
-  const _logger = logs.getLogger('database')
+export async function createDBComponent(components: Pick<AppComponents, 'pg'>): Promise<IDatabaseComponent> {
+  const { pg } = components
 
-  const getCommunityMembers = async (communityId: string) => {
-    const result = await pg.query(
-      SQL`SELECT LOWER(address) as address FROM community_members WHERE community_id = ${communityId}`
-    )
+  const getCommunityMembers = async (
+    communityId: string,
+    options?: {
+      include?: string[]
+      exclude?: string[]
+    }
+  ) => {
+    const includeAddresses = options?.include?.map((address) => `'${address}'`)
+    const excludeAddresses = options?.exclude?.map((address) => `'${address}'`)
+
+    const query = SQL`
+      SELECT LOWER(cm.address) as address FROM community_members cm
+        WHERE cm.community_id = ${communityId}`
+
+    if (includeAddresses) {
+      query.append(` AND cm.address IN (${includeAddresses.join(',')})`)
+    }
+
+    if (excludeAddresses) {
+      query.append(` AND cm.address NOT IN (${excludeAddresses.join(',')})`)
+    }
+
+    // TODO: Filter kicked and banned addresses
```

Comment:
Couldn't `excludeAddresses` fill this purpose already? or do you want to ensure that business logic directly on database side?

Reply:
I think it should be on the database side. I haven't added the filter on the where because we don't have any tables related to communities yet 🫣